### PR TITLE
fixes #18631 - display future dated subscriptions

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by thomasmckay** [comment]
FYI http://projects.theforeman.org/issues/18811 - search for future dates non-intuitive / broken

**Comment by jlsherrill** [comment]
LGTM, but do you have a manifest i could test with?

**Comment by jlsherrill** [comment]
I can't seem to add future subs to a content host's subscriptions.   (they don't show up)

I'm also not able to add ANY subscriptions to an activation key, but i'm seeing that in master.  Will investigate a bit.

**Comment by jlsherrill** [comment]
After figuring out my akey issue, future subs are not showing up there either.

**Comment by jlsherrill** [comment]
Also, don't forget to sqash your commits 🍌

**Comment by thomasmckay** [comment]
@jlsherrill - ready to go, i think

File: app/lib/katello/resources/candlepin.rb [hunk 1]
```diff
@@ -399,12 +399,15 @@ def imports(organization_name)
           end
 
           def pools(owner_label, filter = {})
+            params = hash_to_query(filter)
+            params += params == '?' ? '' : '&'
+            params += "add_future=true"
```
Comment (jlsherrill) [inline app/lib/katello/resources/candlepin.rb]: nit pick, but why not:
```
filter[:add_future] ||= true
params = hash_to_query(filter)
```

File: db/migrate/20170222131211_change_pool_columns_to_dates.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class ChangePoolColumnsToDates < ActiveRecord::Migration
+  def up
+    # 2047-02-14T03:33:40+0000
+    #change_column(:katello_pools, :start_date, :datetime)
+    #change_column(:katello_pools, :end_date, :datetime)
```
Comment (jlsherrill) [inline db/migrate/20170222131211_change_pool_columns_to_dates.rb]: are these comments needed?

**Comment by jlsherrill** [comment]
few minor comments, but otherwise looks good, and works fine

**Comment by thomasmckay** [comment]
@jlsherrill - good nits to pick, updated

**Comment by jlsherrill** [comment]
ACK

**Comment by ehelms** [comment]
Sorry to be the bearer of bad news, but the migration here broke RPM builds [1]. This is the first time we've encountered this and we can correct for the future by adding a sqlite CI job for testing the ability to migrate. But we'll need to fix this as soon as possible to be able to unblock nightly and to pull this change into 3.4. I filed an issue here [2].

[1] http://koji.katello.org/koji/getfile?taskID=621256&name=build.log
[2] http://projects.theforeman.org/issues/19189

**Pull request merged**